staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher.go

```diff
@@ -85,6 +95,22 @@ type watchChan struct {
 	incomingEventChan chan *event
 	resultChan        chan watch.Event
 	errChan           chan error
+	// initialEventsEndBookmarkSent helps us keep track
+	// of whether we have sent an annotated bookmark event.
+	//
+	// it's important to note that we don't
+	// need to track the actual RV because
+	// we only send the bookmark event
+	// after the initial list call.
+	//
+	// when this variable is set to false,
+	// it means we don't have any specific
+	// preferences for delivering bookmark events.
+	initialEventsEndBookmarkSent bool
+	// initialEventsEndBookmarkSentMutex protects initialEventsEndBookmarkSent
+	// this field needs to be protected because there are two
+	// goroutines accessing it.
+	initialEventsEndBookmarkSentMutex sync.RWMutex
```

Comment:
I didn't look super-carefully into the code below and I might have missed something, but I think that using atomic.Bool instead (for initialEventsEndBookmarkSent) without the mutex would simplify the code.